node/src/Transaction.ts

```diff
@@ -384,6 +387,41 @@ export class BaseTransaction<T extends BaseTransaction<T>> {
         return this.addAndReturn(createDel(keys));
     }
 
+    /**
+     * Serialize the value stored at `key` in a Valkey-specific format and return it to the user.
+     *
+     * See https://valkey.io/commands/dump/ for details.
+     *
+     * @param key - The key of the set.
+     *
+     * Command Response - The serialized value of a set. If `key` does not exist, `null` will be returned.
+     */
+    public dump(key: GlideString): T {
```

Comment:
This requires `Bytes` decoder, but it is set in `exec` call, not here.
I think you need to store a flag in transaction and check it in `exec` and raise a error in case of conflict.